docs/docs/adding-app-and-website-functionality.md

```diff
@@ -0,0 +1,78 @@
+
+It's this last phase that bridges the gap between static sites and full-fledged applications. In this phase you can make calls for dynamic data, authenticate users, and perform all the app-like functionality you desire because the page is running a React application.
+
+## Common patterns for Gatsby apps
+
+There are different options for organizing how your pages are created and what they will be responsible for. These patterns can be combined and tweaked for specific use cases like to pull in data at [build time](/docs/glossary#build) for great performance, or call for data at [runtime](/docs/glossary#runtime) for a more dynamic experience.
+
+Because all Gatsby pages are hydrated into React, **any of the following patterns are capable of app-like behavior**. This section is to help explain how people commonly think about Gatsby.
+
+### Static pages
+
+Static files are output by running `gatsby build` from exported components in your `src/pages` folder or from pages created using the [`createPage` API](/docs/node-apis/#createPages), like is shown in this diagram:
+
+![Static Site diagram with pages created from Gatsby automatically and programmatically](./images/simple-static-site.png)
+
+The diagram illustrates the 2 main methods for creating pages in your site:
+
+1. Automatically through `src/pages`
+2. Programmatically with the `createPage` API
+
+_**Note**: plugins and themes can also implement the `createPage` API and create pages on your behalf_
+
+By adding exporting a React component from a file at `src/pages/home` Gatsby can automatically create a static page for you. By looping through markdown files in your filesystem you could create pages for all blog posts programmatically. The docs have more informataion about [creating and modifying pages](/docs/creating-and-modifying-pages/).
+
+These created pages _could_ run JavaScript once React hydrates them, but they don't need to.
+
+### Hybrid app pages
+
+Your created pages can make calls to external services and APIs in order to allow more interactive and dynamic behavior. These are sometimes referred to as hybrid app pages because they share a combination of static features that Gatsby creates to help load your site quickly and performantly as well as traditional web app features.
+
+The following diagram shows a similar site to the one in the previous example, but with a subscription form added to the home page that makes `POST`'s to a database somewhere, as well as blog posts that `GET` recommendations from a database or other API.
+
+![Hybrid Site diagram with pages hitting other services with JavaScript](./images/simple-hybrid-site.png)
+
+Following a pattern like this means you are relying on a backend to remain operational for the features like email signups and blog recommendations, but because the static assets created aren't generated by the server on demand, the content on your site (like your blog posts or home page) will never go down and become unavailable.
```

Comment:
```suggestion
Following a pattern like this means you are relying on a backend to remain operational for features like email signups and blog recommendations, but because the static assets created aren't generated by the server on demand, the content on your site (like your blog posts or home page) will never go down and become unavailable.
```